public/advanced.protocol

```diff
@@ -0,0 +1,216 @@
+      this.props.closeModal(MODAL_NEW_NODE);
+    }
+  },
+
+  render: function() {
+     var label = function (node) { return node.nickname; }
+     var form = this.props.config.params.form;
+
+     return createElement('div', { className: "name-generator" },
+      [
+        createElement('div', { className: "name-generator__main" },
+          createElement('div', { className: "name-generator__nodes" },
+            [
+              createElement('h1', null, 'My Amazing Interface'),
+              createElement(api.elements.NodeList, {
+                network: this.props.activeOriginNetwork,
+                label: label
+              })
+            ]
+          )
+        ),
+        createElement(api.elements.Modal, { name: MODAL_NEW_NODE, title: form.title },
+          createElement(api.elements.Form, Object.assign({ form: form.formName, onSubmit: this.handleAddNode }, form))
+        ),
+        createElement('button', { className: "name-generator__add-person", onClick: function() { this.props.openModal(MODAL_NEW_NODE); }.bind(this) },
+          'Add a person'
+        )
+      ]
+    );
+  }
+});
+
+function mapStateToProps(state) {
+  return {
+    newNodeAttributes:  api.selectors.session.newNodeAttributes(state),
+    activeOriginNetwork: api.selectors.network.activeOriginNetwork(state),
+  };
+}
+
+function mapDispatchToProps(dispatch) {
+  return {
+    addNode: bindActionCreators(api.actions.network.addNode, dispatch),
+    closeModal: bindActionCreators(api.actions.modal.closeModal, dispatch),
+    openModal: bindActionCreators(api.actions.modal.openModal, dispatch),
+  };
+}
+
+MyAmazingInterface = environment.connect(mapStateToProps, mapDispatchToProps)(MyAmazingInterface);
+
+console.log('I was run 5');
+
+return {
+  config: {
+    "name": "My first interview protocol",
+    "version": "1.2.3",
+    "required": "1.2.4",
+    "exportPath": "some/path/here.json",
+    "data": {
+      "previous": {
+        nodes: [
+          {
+            uid: "previous_1",
+            type: "person",
+            name: "Fred",
+            nickname: "Foo",
+          },
+          {
+            uid: "previous_2",
+            type: "person",
+            name: "Bob",
+            nickname: "Bar",
+          },
+          {
+            uid: "previous_3",
+            type: "person",
+            name: "Barry",
+            nickname: "Baz",
+          }
+        ]
+      },
+      "foodNetwork": {
+        nodes: [
+          {
+            type: "food",
+            subType: "vegetable",
+            name: "tomato",
+          },
+          {
+            type: "food",
+            subType: "fruit",
+            name: "lime",
+          },
+          {
+            type: "food",
+            subType: "fruit",
+            name: "bluefruit",
+          }
+        ]
+      }
+    },
+    "stages": [
+      {
+        "id": "namegen1",
+        "type": {
+          "custom": MyAmazingInterface,
```

Comment:
@rebeccamadsen What do you think the protocol should look like for this? We could make it so that type was always an object, i.e: `type: { preset: 'name-generator' }`, or we could keep this short hand. Either way, I'm happy to implement the fix to menu.